notary-server/src/service.rs

```diff
@@ -76,7 +77,23 @@ pub async fn upgrade_protocol(
     let session_id = params.session_id;
     // Fetch the configuration data from the store using the session_id
     let max_transcript_size = match notary_globals.store.lock().await.get(&session_id) {
-        Some(max_transcript_size) => max_transcript_size.to_owned(),
+        Some(data) => {
+            // Check if the session id is still alive
```

Comment:
Since we are relying on session id to prevent DOS on the /notarize endpoint, I added this expiration mechanism to prevent reuse of the same id